packages/clay-slider/src/index.tsx

```diff
@@ -0,0 +1,147 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import React, {useEffect, useRef, useState} from 'react';
+
+interface IProps extends React.InputHTMLAttributes<HTMLInputElement> {
+	/**
+	 * Flag that will disable component features.
+	 */
+	disabled?: boolean;
+
+	/**
+	 * Sets maximum permitted value.
+	 */
+	max?: number;
+
+	/**
+	 * Sets minimum permitted value
+	 */
+	min?: number;
+
+	/**
+	 * Callback will always be called when the slider value changes.
+	 */
+	onValueChange: (value: number) => void;
+
+	/**
+	 * Flag to show tooltip or not.
+	 */
+	showTooltip?: boolean;
+
+	/**
+	 * Sets stepping interval.
+	 */
+	step?: number;
+
+	/**
+	 * Set tooltip position.
+	 */
+	tooltipPosition?: 'top' | 'bottom';
+
+	/**
+	 * Set the current value of the slider.
+	 */
+	value: number;
+}
+
+const THUMB_WIDTH = 24;
+
+const calcProgressWidth = (
+	element: HTMLInputElement,
+	value: number,
+	min: number,
+	max: number,
+	step: number
+) => {
+	const rangeWidth = element.clientWidth;
+	const currentStep = (value - min) / step;
+	const totalSteps = (max - min) / step;
+	const progressWidth = (currentStep / totalSteps) * 100;
+	const ratio =
```

Comment:
Make sure we have new lines separating non-sortable items. For example, since line `progressWidth ` is dependent on `totalSteps`, they should have a new line between.

Reply:
Makes sense, thanks!